### PR TITLE
add lsp-haskell and some hook to trigger lsp with haskell-mode.

### PR DESCRIPTION
**Pull request by hughjfchen**

### CODE REVIEW

File: elisp/init-haskell.el [hunk 1]
```diff
@@ -38,6 +38,17 @@
 ;; HaskellModePac
 (use-package haskell-mode
   :mode "\\.hs\\'")
+
+;; Install and configure lsp-haskell
+(use-package lsp-haskell
+  :ensure t
```
Comment (MatthewZMD) [inline elisp/init-haskell.el:44]: `:ensure t` is unneccesary

File: elisp/init-haskell.el [hunk 2]
```diff
@@ -38,6 +38,17 @@
...
+  :config
+  (setq lsp-haskell-server-path "haskell-language-server")
+  )
```
Comment (MatthewZMD) [inline elisp/init-haskell.el:47]: Please put this on the same line

File: elisp/init-haskell.el [hunk 3]
```diff
@@ -38,6 +38,17 @@
...
+
+(add-hook 'haskell-mode-hook #'lsp)
+(add-hook 'haskell-literate-mode-hook #'lsp)
```
Comment (MatthewZMD) [inline elisp/init-haskell.el:50]: Use `:hook` instead, please refer to my other config.

**Review by MatthewZMD — CHANGES_REQUESTED** [review]
Thank you for your contribution!! Please see my notes for changes.

File: elisp/init-haskell.el (hunk 1 above)
Comment (hughjfchen) [reply elisp/init-haskell.el:44]: Oh,yes.

File: elisp/init-haskell.el (hunk 2 above)
Comment (hughjfchen) [reply elisp/init-haskell.el:47]: Like following?
`   :config (setq lsp-haskell-server-path "haskell-language-server"))`

File: elisp/init-haskell.el (hunk 3 above)
Comment (hughjfchen) [reply elisp/init-haskell.el:50]: sure.

File: elisp/init-haskell.el (hunk 2 above)
Comment (MatthewZMD) [reply elisp/init-haskell.el]: Yes

**Comment by MatthewZMD** [comment]
Thank you for the contribution 😉

**Pull request merged**